service-base/src/test/java/org/eclipse/hono/service/resourcelimits/PrometheusBasedResourceLimitChecksTest.java

```diff
@@ -90,11 +97,20 @@ public void setup() {
         cacheProvider = mock(CacheProvider.class);
         when(cacheProvider.getCache(any())).thenReturn(limitsCache);
 
+        spanContext = mock(SpanContext.class);
+
+        span = mock(Span.class);
+        when(span.context()).thenReturn(spanContext);
+        final Tracer.SpanBuilder spanBuilder = mockSpanBuilder(span);
+
+        tracer = mock(Tracer.class);
```

Comment:
unless you also want to verify that Spans are created and finished, you will probably be better off with simply using the `NoopTracer` here ...

Reply:
Now `mock(NoopTracer.class)` is used instead of `mock(Tracer.class)`.